UI improvements and style panel fixes in the email editor

costasovo opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

- Updates texts in the modal for sending test emails
- Updates texts in the modal displayed before switching to the template mode and unifies buttons on both modals (the first modal opens when a user clicks on the template block in canvas, the second modal whena  user attempts to switch to template mode from sidebar's template dropdown)
- Fixes the style preview component displayed at the top of the styles panel. The component didn't update when user changed styles
- Fixes that when selecting colors in Style > Colors the theme palette was not present
- Removed hardcoded black heading color. A user can now unset the header color, and headings will use text color instead. This improves alignment with Styles in the site editor

Closes WOOPLUG-4200.

### Screenshots or screen recordings:

<!-- If this PR includes UI changes, please provide screenshots or a screen recording for clarity. -->
<!-- This section can be removed if not applicable. -->

| Before | After |
| ------ | ----- |
|   ![Screenshot 2025-05-16 at 15 42 52](https://github.com/user-attachments/assets/cf5191eb-c043-4de7-943a-0105d39ae585) |  ![Screenshot 2025-05-16 at 15 41 29](https://github.com/user-attachments/assets/b82d38bc-a3ca-4682-8822-413c2e9bfe13)   |
|  ![Screenshot 2025-05-16 at 15 44 40](https://github.com/user-attachments/assets/32412d67-deaf-4741-931d-36a1969a6b92) | ![Screenshot 2025-05-16 at 15 47 35](https://github.com/user-attachments/assets/48d02db4-c2a5-49a5-9539-b0e9d1a7c145) |
| ![Screenshot 2025-05-16 at 15 48 24](https://github.com/user-attachments/assets/1b192111-2faa-471e-8ead-3a9f50c64a38) | ![Screenshot 2025-05-16 at 15 48 11](https://github.com/user-attachments/assets/7043ffbe-56f0-4e52-8de7-cdaccc292d1f) |

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the block email editor WooCommece > Settings > Advanced > Features
2. Open an email in the editor WooCommece > Settings > Emails
3. Verify the updated features listed in the chnages section work as expected

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

costasovo committed:
Update texts in preview modal

WOOPLUG-4200

costasovo committed:
Address deprecation warnings from the TextControl in the preview modal

- opt-in to the new default input height which matches the design
- opt-in to the new default no bottom margin. This also required adding
the margin using CSS to match the design.

WOOPLUG-4200

costasovo committed:
Update switch to template mode modals

The modal opened by clicking on canvas is handled by core, so I chose to use translation
hooks to replace the copy.
WOOPLUG-4200

costasovo committed:
Fix the styles preview component to reflect changes

The main issue was that the Preview component was using static settings
value when reading style, which was set by the server.

I updated it to use live styles from the useEmailStyles hook, but I had to do additional
changes to be able to get values when the style was set as CSS variable.

WOOPLUG-4200

costasovo committed:
Fix that the theme color palette was not available in style > color settings

The color selection component uses theme settings to generate color palettes.
The site palette was added only to editor settings but was not present in the base theme,
which is used for the color selection component in the ScreenColors component.

This commit moves the logic for extracting handpicked site theme settings and styles
to a proper place.

WOOPLUG-4200

costasovo committed:
Remove hardcoded heading color default

This improves alignment with the site's global styles.
Heading color can be unset, and the heading will use the text color.
In case there is an extra color set for the headings, it is set
by WordPress's style system on the nested h1 element.

This also fixes an issue with Outlook, which was ignoring the color
set on the main wrapper.

WOOPLUG-4200

costasovo committed:
Add changelogs

WOOPLUG-4200

costasovo committed:
Remove duplicate declaration for EmailStyles

WOOPLUG-4200

lysyjan reviewed (commented):
@costasovo Thank you for nice fixes and improvements 👍 

I have noticed an issue with the preview styles component.
When I changed all colors, the colors of the dots in the preview were still black and white.

I am sharing some images for better understanding:
<img width="288" alt="Screenshot 2025-05-19 at 16 31 41" src="https://github.com/user-attachments/assets/5b0a8252-cb53-40c9-af10-5e8a1eb7d309" />
<img width="284" alt="Screenshot 2025-05-19 at 16 31 31" src="https://github.com/user-attachments/assets/e7e1f1a8-0db8-4516-83bd-09c81f63639f" />

When I tested the site editor, it worked as expected. Do you think we should address it in this PR or should we create a new ticket?

costasovo commented:
> When I tested the site editor, it worked as expected. Do you think we should address it in this PR or should we create a new ticket?

Nice catch! @lysyjan I kept the same logic as we had before and I thought it is the same asi in [core's global styles](https://github.com/WordPress/gutenberg/blob/7fa03fafeb421ab4c3604564211ce6007cc38e84/packages/edit-site/src/components/global-styles/hooks.js#L68-L73). 
The two dots are currently the two main colors from pallette and if some of them is chosen for text, heading or background it is replaced by the next one.

But I can see that the behavior in the site editor differs and it looks like the upper dot now uses the text color. I'll take another look. No need for another ticket.

costasovo committed:
Update preview component highlight colors selection logic

This commit updates the logic for selection of highlighted
colors in the preview component to be aligned with the updated
logic in the site editor.

WOOPLUG-4200

costasovo commented:
@lysyjan I updated the logic for preview colors following the most recent code in Gutenberg. See https://github.com/woocommerce/woocommerce/pull/58078/commits/91f5b022b7267085ad1712622f514229835822ef
Note that to be able to define color for H1, you need to use the more menu (...) in the colors section.
<img width="543" alt="Screenshot 2025-05-20 at 13 01 46" src="https://github.com/user-attachments/assets/d90911e9-cac6-4e8d-a7d6-b6627b28312d" />

lysyjan reviewed (approved):
Thank you for those fixes @costasovo 👍 

I tested this PR locally with WP 6.8.1 and theme Twenty-Twenty Five.
- I tried updating colors and the color preview component.
- I checked all new texts and the template mode.

<img width="465" alt="Screenshot 2025-05-20 at 17 27 22" src="https://github.com/user-attachments/assets/ef3d9df5-25fc-4a70-88c3-3d7739e1fd00" />

<img width="521" alt="Screenshot 2025-05-20 at 17 27 36" src="https://github.com/user-attachments/assets/ede5aaa1-8dae-4ede-8b7c-9e49b900de48" />


<img width="144" alt="Screenshot 2025-05-20 at 17 30 06" src="https://github.com/user-attachments/assets/72032700-23fc-49e3-aee6-e450a4169cc7" />
